### PR TITLE
Models

### PR DESCRIPTION
**Pull request by 15lnestor**
## Description of Change
Adds the division and team model to the application. Also includes basic validation specs. Fixes comments on #1 after it was merged. 

## Alternate Designs
N/A

## Benefits
N/A

## Possible Drawbacks
N/A

### CODE REVIEW

**Comment by 15lnestor** [comment]
How long do you want PRs to be? I've read that 10 files is a good amount to aim for, but right now everything I am doing is pretty trivial. I've been dividing it up like this so you aren't bombarded with a bunch of code at once.

File: spec/models/division_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+RSpec.describe Division, type: :model do
+  let(:division) { FactoryGirl.create(:division) }
+
+  it { expect(division).to have_many(:teams).dependent(:destroy) }
+  it { is_expected.to validate_presence_of(:name) }
+  it { is_expected.to validate_length_of(:name) }
```
Comment (scottmascio2115) [inline spec/models/division_spec.rb]: This looks really good I would expand on the validation and ensure the length is between 6 and 50. 

I think you can do [this](https://github.com/thoughtbot/shoulda-matchers/blob/master/lib/shoulda/matchers/active_model/validate_length_of_matcher.rb#L121)

```
             it do
               should validate_length_of(:password).
                 is_at_least(5).is_at_most(30)
             end
```

File: spec/factories/leagues.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 FactoryGirl.define do
   factory :league do
-    name "New League"
+    name "Division Name"
```
Comment (scottmascio2115) [inline spec/factories/leagues.rb]: I think this should be `"League Name"`

File: app/models/division.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Division < ApplicationRecord
+  belongs_to :league
```
Comment (scottmascio2115) [inline app/models/division.rb]: This looks good check out adding the inverse of option as well [here](https://www.viget.com/articles/exploring-the-inverse-of-option-on-rails-model-associations)

File: Gemfile [hunk 4]
```diff
@@ -39,6 +41,7 @@ end
 group :test do
   gem 'cucumber-rails', :require => false
   gem 'database_cleaner' # database_cleaner is not required, but highly recommended
+  gem 'shoulda-matchers', git: 'https://github.com/thoughtbot/shoulda-matchers.git', branch: 'rails-5'
```
Comment (scottmascio2115) [inline Gemfile]: Awesome!

File: Gemfile [hunk 5]
```diff
@@ -21,8 +21,10 @@ gem 'jbuilder', '~> 2.5' # Build JSON APIs with ease. Read more: https://github.
 # gem 'capistrano-rails', group: :development # Use Capistrano for deployment
 
 gem 'figaro'
-# gem 'haml' # Do we need this?
-#gem "haml-rails", "~> 0.9"
+
+# Commented to prevent deprecation messages
```
Comment (scottmascio2115) [inline Gemfile]: We should figure out what you wanna do with Haml at some point. Not important now

File: app/models/division.rb (hunk 3 above)
Comment (15lnestor) [reply app/models/division.rb]: It looks like Rails automatically adds the inverse_of option when the class name is the same as the association name. [PR](https://github.com/rails/rails/pull/9522/files)

> As of version 4.1, Rails will try to automatically set the :inverse_of option for you

Is it still good to explicitly state the inverse of option when the class/association names are the same?

Comment (scottmascio2115) [reply app/models/division.rb]: Nope good catch. As long as you know what the inverse option does feel free to skip adding it

**Pull request merged**